core/identity-management/iin-agent/src/protocols/externalOperations.ts

```diff
@@ -5,26 +5,224 @@
 
+// key: get using function "getSecurityDomainMapKey"
+export let securityDomainMap = new Map<string, string | iin_agent_pb.AttestedMembership>();
+export let counterAttestationsMap = new Map<string, string | iin_agent_pb.CounterAttestedMembership>();
+
+// key: Nonce
+export let foreignAgentResponseCount = new Map<string, { current: number, total: number }>();
+export let localAgentResponseCount = new Map<string, { current: number, total: number }>();
+
+export function getSecurityDomainMapKey(securityDomain: string, iinAgent: string, nonce?: string | undefined) {
+    if (nonce) {
+        return 'SEC_DOM_MAP:' + securityDomain + ':' + iinAgent + ':' + nonce;
+    } else {
+        return 'SEC_DOM_MAP:' + securityDomain + ':' + iinAgent;
+    }
+}
 
 // Handles communication with foreign IIN agents
-export const syncExternalStateFromIINAgent = async (securityDomainUnit: iin_agent_pb.SecurityDomainMemberIdentity) => {
-    console.log('syncExternalStateFromIINAgent:', securityDomainUnit.getSecurityDomain(), '-', securityDomainUnit.getMemberId());
+export const syncExternalStateFromIINAgent = async (remoteSecurityDomainUnit: iin_agent_pb.SecurityDomainMemberIdentity, securityDomain: string, memberId: string) => {
+    const remoteSecurityDomain = remoteSecurityDomainUnit.getSecurityDomain();
+    const nonce = uuidv4();
+    console.log('syncExternalStateFromIINAgent:', remoteSecurityDomain, '-', remoteSecurityDomainUnit.getMemberId(), ' nonce:', nonce);
+    
+    // const sleepTime = 2000 // 2 seconds
+    // const timeoutTime = 300 * 1000 // 5 minutes
+    // const timeoutIterations = timeoutTime/sleepTime // 5 minutes
+    
+    const requestingSecurityDomain = new iin_agent_pb.SecurityDomainMemberIdentity();
+    requestingSecurityDomain.setSecurityDomain(securityDomain);
+    requestingSecurityDomain.setMemberId(memberId);
+    
+    const request = new iin_agent_pb.SecurityDomainMemberIdentityRequest();
+    request.setRequestingNetwork(requestingSecurityDomain);
+    request.setNonce(nonce);
+    
+    // Request Attested Membership from all members of remote security_domain
+    const remoteSecurityDomainDNS = utils.getSecurityDomainDNS(remoteSecurityDomain);
+    let c = 0;
+    for (const iinAgent in remoteSecurityDomainDNS) {
+        const iinAgentClient = utils.getIINAgentClient(remoteSecurityDomain, iinAgent, remoteSecurityDomainDNS);
+        remoteSecurityDomainUnit.setMemberId(iinAgent);
+        request.setSourceNetwork(remoteSecurityDomainUnit);
+        console.log(`Requesting attested memberships from: ${remoteSecurityDomain} - ${iinAgent}`);
+        iinAgentClient.requestIdentityConfiguration(request, utils.defaultCallback);
+        // remoteSecDomMapKeys.push(getSecurityDomainMapKey(remoteSecurityDomain, iinAgent, nonce));
+        c++;
+    }
+    foreignAgentResponseCount.set(nonce, { current:0, total: c});
 };
 
-// Generates security domain unit's state/configuration
+// Generates network unit's state/configuration
```

Comment:
Wasn't the earlier label more accurate?

Reply:
Oh I didn't change this, may have been due to git merge probably retained old change. Will make it security domain.